rir/src/compiler/opt/force_dominance.cpp

```diff
@@ -440,10 +440,8 @@ bool ForceDominance::apply(Compiler&, ClosureVersion* cls, Code* code,
             if (auto f = Force::Cast(*ip)) {
                 if (auto mkarg = MkArg::Cast(f->followCastsAndForce())) {
                     if (mkarg->isEager()) {
-                        anyChange = true;
-                        Value* eager = mkarg->eagerArg();
-                        f->replaceUsesWith(eager);
-                        next = bb->remove(ip);
+                        assert(false &&
```

Comment:
I don't see why we would want this invariant? I mean this just means that the cleanup pass was not run, which should not be a prerequisite for this pass to work.